spec/lib/user_segments_spec.rb

```diff
@@ -0,0 +1,176 @@
+require 'rails_helper'
+
+describe UserSegments do
+
+  describe "#all_users" do
+    it "returns all active users with newsletter enabled" do
+      active_user1 = create(:user, newsletter: true)
+      active_user2 = create(:user, newsletter: true)
+      active_user3 = create(:user, newsletter: false)
+      erased_user  = create(:user, erased_at: Time.current)
+
+      expect(UserSegments.all_users).to include active_user1
+      expect(UserSegments.all_users).to include active_user2
+      expect(UserSegments.all_users).to_not include active_user3
+      expect(UserSegments.all_users).to_not include erased_user
+    end
+  end
+
+  describe "#proposal_authors" do
+    it "returns users that have created a proposal" do
+      user1 = create(:user)
+      user2 = create(:user)
+
+      proposal = create(:proposal, author: user1)
+
+      proposal_authors = UserSegments.proposal_authors
+      expect(proposal_authors).to include user1
+      expect(proposal_authors).to_not include user2
+    end
+
+    it "does not return duplicated users" do
+      user = create(:user)
+
+      proposal1 = create(:proposal, author: user)
+      proposal2 = create(:proposal, author: user)
+
+      proposal_authors = UserSegments.proposal_authors
+      expect(proposal_authors).to contain_exactly(user)
+    end
+  end
+
+  describe "#investment_authors" do
+    it "returns users that have created a budget investment" do
+      user1 = create(:user)
+      user2 = create(:user)
+
+      investment = create(:budget_investment, author: user1)
+      budget = create(:budget)
+      investment.update(budget: budget)
+
+      investment_authors = UserSegments.investment_authors
+      expect(investment_authors).to include user1
+      expect(investment_authors).to_not include user2
+    end
+
+    it "does not return duplicated users" do
+      user = create(:user)
+
+      investment1 = create(:budget_investment, author: user)
+      investment2 = create(:budget_investment, author: user)
+      budget = create(:budget)
+      investment1.update(budget: budget)
+      investment2.update(budget: budget)
+
+      investment_authors = UserSegments.investment_authors
+      expect(investment_authors).to contain_exactly(user)
+    end
+  end
+
+  describe "#feasible_and_undecided_investment_authors" do
+    it "returns authors of a feasible or an undecided budget investment" do
+      user1 = create(:user)
+      user2 = create(:user)
+      user3 = create(:user)
+
+      investment1 = create(:budget_investment, :feasible, author: user1)
```

Comment:
I see many `create`'s for users and investments in this file repeating on each scenario, I think just with a bunch of `let` at the top we could make scenarios much faster to be read. But at least descriptive naming would be a plus like instead of `investment1` a `feasible_investment` would make much sense